libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1415,18 +1415,16 @@ std::string YulUtilFunctions::cleanUpStorageArrayEndFunction(ArrayType const& _t
 		return Whiskers(R"(
 			if lt(startIndex, len) {
 				// Size was reduced, clear end of array
-				let oldSlotCount := <convertToSize>(len)
 				let newSlotCount := <convertToSize>(startIndex)
 				let arrayDataStart := <dataPosition>(array)
 				let deleteStart := add(arrayDataStart, newSlotCount)
-				let deleteEnd := add(arrayDataStart, oldSlotCount)
 				<?packed>
 					// if we are dealing with packed array and offset is greater than zero
 					// we have  to partially clear last slot that is still used, so decreasing start by one
 					let offset := mul(mod(startIndex, <itemsPerSlot>), <storageBytes>)
 					if gt(offset, 0) { <partialClearStorageSlot>(sub(deleteStart, 1), offset) }
 				</packed>
-				<clearStorageRange>(deleteStart, deleteEnd)
+				<clearStorageRange>(deleteStart, <convertToSize>(sub(len, startIndex)))
```

Comment:
Nitpick: I think it would be better for readability to have 
`let slotCount := <convertToSize>(sub(len, startIndex))`
```suggestion
				<clearStorageRange>(deleteStart, slotCount)
```